pkg/kademlia/kademlia_test.go

```diff
@@ -153,10 +160,12 @@ func TestLookup(t *testing.T) {
 		err := v.k.lookup(context.Background(), v.target, v.opts)
 		assert.Equal(t, v.expectedErr, err)
 	}
-
+	cleanup(ctx, t, k, dir)
```

Comment:
Prefer to use defers for cleanup because that will also try to cleanup when the test fails.